core/src/main/java/bisq/core/dao/state/storage/DaoStateStorageService.java

```diff
@@ -76,10 +76,32 @@ public String getFileName() {
         return FILE_NAME;
     }
 
-    public void requestPersistence(DaoState daoState, LinkedList<DaoStateHash> daoStateHashChain) {
+    public void requestPersistence(DaoState daoState,
+                                   LinkedList<DaoStateHash> daoStateHashChain,
+                                   Runnable completeHandler) {
+        if (daoState == null) {
+            completeHandler.run();
+            return;
+        }
+
         store.setDaoState(daoState);
         store.setDaoStateHashChain(daoStateHashChain);
-        persistenceManager.requestPersistence();
+
+        // We let the persistence run in a thread to avoid the slow protobuf serialisation to happen on the user
+        // thread. We also call it immediately to get notified about the completion event.
+        new Thread(() -> {
+            Thread.currentThread().setName("Serialize and write DaoState");
+            persistenceManager.persistNow(() -> {
+                // After we have written to disk we remove the reference to the daoState. We have in the meantime
+                // already cloned the next daoState and if we would not release the reference we would have 3 times
+                // the daoState data in memory.
+                store.setDaoState(null);
```

Comment:
It looks to me like the store is reset before completehandler is called which is where the new clone of daoState is made. Comment makes it seem like the clone is made before store is cleared.

Reply:
Yes comment was from earlier version. I updated it in a new commit.